### ISSUE TITLE
Should support upper-case headers

### ISSUE THREAD

**Issue by aaronatbissell** [jshttp/type-is#56]
According to [RFC 9110](https://www.rfc-editor.org/rfc/rfc9110.html#name-field-names), I believe headers are meant to be case-insensitive.

> Field names are case-insensitive ...

Running the test suite in this project with a header of `Content-Type` yields errors. Would be nice to case-insensitively process those headers

**Comment by dougwilson** [comment]
It is already case insensitive as Node.js HTTP lower cases them already. Try out the example at https://github.com/jshttp/type-is?tab=readme-ov-file#api and send upper case headers to see.

**Comment by aaronatbissell** [comment]
That's true if you are using Node.js HTTP. We are using this in a project that does not use Node.js HTTP and has some clients that happen to send headers with upper-case characters.

**Comment by dougwilson** [comment]
This project only works with Node.js HTTP. If you are using something else, you should be using something designed to work qith whatever you are using instead.

**Comment by dougwilson** [comment]
From the docs:

> The request argument is expected to be a Node.js HTTP request.

**Comment by aaronatbissell** [comment]
I'm sure you know this already, but `express` uses this package. It's unfortunate that now `express` doesn't work because of a case sensitivity issue with a header.

**Comment by dougwilson** [comment]
Express uses Node.js HTTP. The `req.headers` object that enters into this module onlt contains lower case header names, even when the client sends them in capitals. This is because Node.js HTTP (and thus Express.js) will always lower case the headers when they enter the server.

Perhaps you can provide a full Express.js app and show us a cURL request or similar with upper case headers that does not work.

https://nodejs.org/api/http.html#messageheaders

> The request/response headers object.
> 
> Key-value pairs of header names and values. Header names are lower-cased.

**Comment by aaronatbissell** [comment]
Here's an [issue](https://github.com/CodeGenieApp/serverless-express/issues/347) from another project where they were seeing the exact same problem. I'm confident it's because their `Content-Type` header is capitalized. I'll be the first to admit that this isn't technically going over HTTP because it's invoking a lambda function directly instead of using something like API Gateway, but it's still possible to invoke express with a request that technically satisfies the shape of the `request` object, but has upper-case headers.

**Comment by aaronatbissell** [comment]
I've requested an enhancement to the other library to mimic the behavior of Node.js HTTP

**Comment by blakeembrey** [comment]
I'm going to close this out as the package is designed for node-style HTTP headers, but for anyone following you can use this by passing only `{ headers: {} }` as long as headers is the correct shape (lower-case the keys). It doesn't have to be a full node.js request.
